src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.cs

```diff
@@ -125,28 +125,33 @@ public class TarReader_File_Tests : TarReader_File_Tests_Base
         public void Read_Archive_LongPath_Over255(TarEntryFormat format, TestTarFormat testFormat) =>
             Read_Archive_LongPath_Over255_Internal(format, testFormat);
 
-        [Fact]
-        public void Read_NodeTarArchives_Successfully()
+        [Theory]
+        [InlineData("node-tar")]
+        [InlineData("tar-rs")]
+        public void Read_TestArchives_Successfully(string subset)
         {
-            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "node-tar");
+            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", subset);
             foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
             {
                 using FileStream sourceStream = File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read);
                 using var reader = new TarReader(sourceStream);
 
                 TarEntry? entry = null;
-                while (true)
+                while ((entry = reader.GetNextEntry()) != null)
                 {
-                    Exception ex = Record.Exception(() => entry = reader.GetNextEntry());
-                    Assert.Null(ex);
-
-                    if (entry is null) break;
-
-                    ex = Record.Exception(() => entry.Name);
-                    Assert.Null(ex);
-
-                    ex = Record.Exception(() => entry.Length);
-                    Assert.Null(ex);
+                    Assert.NotNull(entry.Name);
+                    Assert.True(Enum.IsDefined(entry.EntryType));
+                    Assert.True(Enum.IsDefined(entry.Format));
+
+                    if (entry.EntryType == TarEntryType.Directory)
+                        continue;
```

Comment:
This directory entry check is unnecessary if the entry.DataStream is already being checked for null a few lines below.